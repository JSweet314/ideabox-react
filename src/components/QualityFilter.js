import React from 'react';
import PropTypes from 'prop-types';
import '../styles/QualityFilter.css';

const QualityFilter = ({filterByQuality, searchIdeas}) => {
  return (
    <div className="quality-filter">
      <input
        className="search"
        type="search"
        placeholder="SEARCH"
        onChange={(e) => searchIdeas(e.target.value)}
      />
      <label htmlFor="">
        <span>Filter by Idea Quality: </span>  
        <select 
          name="quality"
          className="quality-filter__select"
          onChange={(e) => filterByQuality(e.target.value)}>
          <option value="all">all</option>
          <option value="swill">swill</option>
          <option value="plausible">plausible</option>
          <option value="genius">genius</option>
        </select>
      </label>
    </div>
  );
};

QualityFilter.propTypes = {
  filterByQuality: PropTypes.func.isRequired
};

export default QualityFilter;
