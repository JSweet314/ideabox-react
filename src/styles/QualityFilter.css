.quality-filter {
  font-size: 2rem;
  vertical-align: center;
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: 0.5rem 0 0 0;
}

.search {
  font-family: 'Open Sans', sans-serif;
  margin: 2rem 0 0 0;
  height: 4rem;
  padding: 0 0 0 1rem;
  border-style: unset;
  border: 2px solid lightgray;
  font-size: 2rem;
}

.search::placeholder {
  font-family: 'Open Sans', sans-serif;
}

.quality-filter__select {
  font-size: 1.5rem;
  margin: 0 0 0 0.5rem;
  cursor: pointer;
}