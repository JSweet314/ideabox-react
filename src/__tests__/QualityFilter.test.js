import React from 'react';
import {shallow, mount} from 'enzyme';
import QualityFilter from '../components/QualityFilter.js';

describe('QualityFilter', () => {
  it('should render our good friend, QualityFilter', () => {
    const qualityFilter = shallow(<QualityFilter filterByQuality={jest.fn()} />);
    expect(qualityFilter).toMatchSnapshot();
  });

  it('should call the filterByQuality method when options are changed', () =>{
    const qualityFilter = mount(<QualityFilter filterByQuality={jest.fn()} />);
    const select = qualityFilter.find('.quality-filter__select');

    select.simulate('change', {target: {value: 'swill'}});

    expect(qualityFilter.props().filterByQuality).toHaveBeenCalledTimes(1);
  });
});