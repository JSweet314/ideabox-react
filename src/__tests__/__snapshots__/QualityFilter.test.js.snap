// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QualityFilter should render our good friend, QualityFilter 1`] = `
<div
  className="quality-filter"
>
  <input
    className="search"
    onChange={[Function]}
    placeholder="SEARCH"
    type="search"
  />
  <label
    htmlFor=""
  >
    <span>
      Filter by Idea Quality: 
    </span>
    <select
      className="quality-filter__select"
      name="quality"
      onChange={[Function]}
    >
      <option
        value="all"
      >
        all
      </option>
      <option
        value="swill"
      >
        swill
      </option>
      <option
        value="plausible"
      >
        plausible
      </option>
      <option
        value="genius"
      >
        genius
      </option>
    </select>
  </label>
</div>
`;
